core/cont/inc/RangeCast.h

```diff
@@ -0,0 +1,185 @@
+// Author: Jonas Rembser, CERN, Jul 2021
+
```

Comment:
I'd propose to move this to `core/foundation/inc/ROOT` (IIUC it has no dependencies on anything ROOT?). It should be in `ROOT/` with a `.hxx` extension because it's namespaced and follows the new interface style. The filename should probably be `RRangeCast.hxx`

Reply:
Okay!